src/main/java/org/jabref/logic/net/URLDownload.java

```diff
@@ -171,6 +172,21 @@ public String getMimeType() {
         return "";
     }
 
+    /**
+     * Check the connection by using the HEAD request.
+     * UnirestException can be thrown for invalid request.
+     * @return the status code of the response
+     */
+    public int checkConnection() {
```

Comment:
This is a new method. So, you can work an a proper interface.

Why not?

```suggestion
    public boolean canBeReached() {
```

See Item 69: Use exceptions only for exceptional conditions of [Effective Java](https://www.oreilly.com/library/view/effective-java-3rd/9780134686097/).

You can log the status code - and the exception (debug level). I think, the end user is not interested in the concrete response code, is he?

The mapping of response codes and exceptions to a proper user information is hard. All responses not falling in the class `2xx` are errors. Thus, both the codes and the exceptions should be mapped to a user message. -- To design that properly is too hard. Just go for a boolean response.